### PR TITLE
1218 warehouse state details

### LINKED ISSUE #1218: Create Silver module for Warehouse State Details
Add a new module in Silver for warehouse_state_details

### PR DESCRIPTION
**Pull request by aman-db**
closes #1218 
Added Silver module for Warehouse State Details 

Also made some changes in MultiWorkspaceDeployment for scopes

### CODE REVIEW

**Comment by souravbaner-da** [comment]
@aman-db  how we are handling non uc users here?

**Comment by souravbaner-da** [comment]
@aman-db target_num_clusters is showing as 0

**Comment by souravbaner-da** [comment]
@aman-db  check for warehouseID = 54a32114fd0e2aa0. Though it is running state but the current number of worker is showing as 0.

**Comment by aman-db** [comment]
> @aman-db check for warehouseID = 54a32114fd0e2aa0. Though it is running state but the current number of worker is showing as 0.

done

**Comment by aman-db** [comment]
> @aman-db target_num_clusters is showing as 0

done

**Comment by aman-db** [comment]
> @aman-db how we are handling non uc users here?

done

**Review by souravbaner-da — APPROVED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala [hunk 1]
```diff
@@ -1408,4 +1456,224 @@ trait SilverTransforms extends SparkSessionWrapper {
...
+    val stateDuringRemovalFinal = stateBeforeRemoval.drop("deletion_timestamp")
+      .unionByName(stateDuringRemoval, allowMissingColumns = true)
+      .select(columns.map(col): _*)
+
+    val warehouseEventsBaselineFinal = warehouseEventsBaseline.join(stateDuringRemovalFinal,Seq("warehouse_id","timestamp"),"anti")
+      .select(columns.map(col): _*)
+      .unionByName(stateDuringRemovalFinal, allowMissingColumns = true)
+
+    warehouseEventsBaselineFinal
+      .withColumn("counter_reset",
```
Comment (gueniai) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1653]: This column name is difficult to understand what it's referring to. Even if this is a table we don't expect customers query directly, it shouldn't be so cryptic.

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala [hunk 2]
```diff
@@ -1408,4 +1456,224 @@ trait SilverTransforms extends SparkSessionWrapper {
...
+        when(
+          lag('state, 1).over(stateUnboundW).isin("STOPPING","TERMINATING", "RESTARTING", "EDITED","TERMINATING_IMPUTED") ||
+            !'isRunning, lit(1)
+        ).otherwise(lit(0))
+      )
+      .withColumn("reset_partition", sum('counter_reset).over(stateUnboundW))
```
Comment (gueniai) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1659]: Same comment as before, this column name is hard to decipher what it is

**Review by gueniai — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala (hunk 1 above)
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala]: done

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala (hunk 2 above)
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala]: done

**Pull request merged**